include/x86/serial.h

```diff
@@ -0,0 +1,11 @@
+#ifndef SERIAL_H
+#define SERIAL_H
+
+#include <stdint.h>
+
+#define SERIAL_PORT 0x3f8
+
+int uart_init();
```

Comment:
You might consider reusing some of the `DEBUG_UART` stuff. See `include/printf.h`. It already declares these functions and maps them to `wolfBoot_printf`

Reply:
refactored `serial.c` in `hal/x86_uart.c` to better match wolfboot uart hal